Source/LibationFileManager/InteropFactory.cs

```diff
@@ -115,12 +117,28 @@ private static List<ProcessModule> LoadModuleList(string exeName)
         private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
         {
             // e.g. "System.Windows.Forms, Version=6.0.2.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
-            var asmName = args.Name.Split(',')[0];
+            var asmName = args.Name.Split(',')[0] + ".dll";
 
-            // `First` instead of `FirstOrDefault`. If it's not present we're going to fail anyway. May as well be here
-            var module = ModuleList.First(m => m.ModuleName.StartsWith(asmName));
+#if DEBUG
+
+            var modulePath = ModuleList.SingleOrDefault(m => m.ModuleName.EqualsInsensitive(asmName))?.FileName;
+#else
+            var here = Path.GetDirectoryName(Environment.ProcessPath);
+
+            // find '*ConfigApp.dll' files
+            var modulePath =
+                Directory.EnumerateFiles(here, asmName, SearchOption.TopDirectoryOnly)
+                .SingleOrDefault();
+
+#endif
+            if (modulePath is null)
+            {
+                //Let the runtime handle any dll not found exceptions.
+                Serilog.Log.Logger.Error($"Unable to load module {args.Name}");
+                return null;
```

Comment:
I assume the program will crash when it gets this null, right? I like that we log it first.

Reply:
Not necessarily. The runtime will attempt to load assemblies for types when those types enter scope. If the code bringing a type into scope is non-critical and wrapped inside a try/catch block, the failure will be caught and execution will continue.